File: 99-Continue/155/level_order_traversal.py
from collections import deque

class Node:
    def __init__(self,data:int):
        self.left = None
        self.right = None
        self.data = data

class BinaryTree:
    def __init__(self):
        self.root = None

    def buildTree(self, n):
        values = list(map(int, input().split()))
        self.root = Node(values[0])
        queue = deque([self.root])
        index = 1
        while queue and index < n:
            current = queue.popleft()
            if index < n:
                current.left = Node(values[index])
                queue.append(current.left)
                index += 1

            if index < n:
                current.right = Node(values[index])
                queue.append(current.right)
                index += 1

    def levelOrderTraversal(self):
        if self.root == None:
            return 

        queue = deque([self.root])
        while queue:
            current = queue.popleft()
            print(current.data,end=" ")
            if current.left:
                queue.append(current.left)
            if current.right:
                queue.append(current.right)

def main():
    tree = BinaryTree()
    n = int(input())
    tree.buildTree(n)
    tree.levelOrderTraversal()

if __name__=="__main__":
    main()

File: 99-Continue/155/LevelOrderTraversal.java
import java.util.*;

public class LevelOrderTraversal {
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        BinaryTree tree = new BinaryTree();
        
        tree.buildTree(n, sc);
        tree.levelOrderTraversal();
        sc.close();
    }
}

class Node{
        Node left,right;
        int data;
        Node(int val){
            this.data = val;
        }
    }
    
    class BinaryTree{
        Node root;
        BinaryTree(){
            root = null;
        }
        
        Node buildTree(int n,Scanner sc){
            if(n==0){
                return null;
            }
            
            root = new Node(sc.nextInt());
            int count = 1;
            Queue<Node> queue = new LinkedList<>();
            queue.offer(root);
            
            while(count<n){
                Node current = queue.poll();
                if(count<n){
                    current.left = new Node(sc.nextInt());
                    queue.offer(current.left);
                    count++;
                }
                if(count<n){
                    current.right = new Node(sc.nextInt());
                    queue.offer(current.right);
                    count++;
                }
            }
            return root;
        }
        
        void levelOrderTraversal(){
            Queue<Node> queue = new LinkedList<>();
            queue.offer(root);
            while(!queue.isEmpty()){
                Node current = queue.poll();
                System.out.print(current.data+" ");
                if(current.left!=null){
                    queue.offer(current.left);
                }
                if(current.right!=null){
                    queue.offer(current.right);
                }
            }
        }
    }